ampd/src/event_sub.rs

```diff
@@ -112,32 +118,31 @@ impl<T: TmClient + Sync> EventPublisher<T> {
     }
 
     async fn process_block(&self, height: block::Height) -> Result<(), EventSubError> {
+        // skip processing if there are no subscribers
+        if self.tx.receiver_count() == 0 {
+            return Ok(());
+        }
+
         let events = iter::once(Event::BlockBegin(height))
-            .chain(
-                self.events(height)
-                    .await?
-                    .into_iter()
-                    .map(|event| event.try_into())
-                    .collect::<Result<Vec<_>, _>>()
-                    .change_context(EventSubError::Publish)?,
-            )
+            .chain(self.events(height).await?.into_iter())
             .chain(iter::once(Event::BlockEnd(height)));
 
         for event in events {
-            self.tx.send(event).change_context(EventSubError::Publish)?;
+            // ignore the error if there are no subscribers
+            let _ = self.tx.send(event);
```

Comment:
it skips processing if no subscribers earlier, why suppress the error here?

Reply:
The previous check prevents the block from being downloaded if no subscriber exists. However, subscribers can still drop while the publisher is in this loop broadcasting events.